src/gui/tray/usermodel.cpp

```diff
@@ -697,6 +702,32 @@ bool User::hasActivities() const
     return _account->account()->capabilities().hasActivities();
 }
 
+QColor User::headerColor() const
+{
+    const auto serverColor = _account->account()->capabilities().serverColor();
+    return serverColor.isValid() ? serverColor : Theme::defaultColor();
+}
+
+QColor User::headerTextColor() const
+{
+    return _account->account()->capabilities().serverTextColor();
+}
+
+QColor User::accentColor() const
+{
+    // This will need adjusting when dark theme is a thing
+    const auto serverColor = _account->account()->capabilities().serverColor();
+
+    if(!serverColor.isValid()) {
+        return QColor();
+    }
+
+    auto darknessAdjustment = (int)((1 - Theme::getColorDarkness(serverColor)) * 8);
+    darknessAdjustment *= darknessAdjustment; // Square the value to pronounce the darkness more in lighter colours
+    const auto adjusted = Theme::isDarkColor(serverColor) ? serverColor : serverColor.darker(125 + darknessAdjustment);
+    return adjusted;
+}
+
```

Comment:
please move this into `Account` and just act as a proxy here
see for example `User::avatar()`
that would remove the need to introduce `Account::capabilitiesChanged` that is in fact not triggering a real refresh of all things stored into server capabilities
I think that properly refreshing everything depending on server capabilities is out of scope for this PR

Reply:
We still need to listen to the signals, otherwise we won't have the updated capabilities which will mean the tray colour will remain in the default blue.

I'll move the guts of the colour acquisition code to the account class, but we will need to keep the signal connections